### ISSUE TITLE
getShortTermLoudnessAtSamplePosition from liveinput (getUserMedia)

### ISSUE THREAD

**Issue by infantilo** [SebastianZimmer/LoudEv#1]
It would be pretty useful to be able to measure loudness on liveinputs.
But haven't found an solution right no as the scriptingnode offers too little buffersize.
Ony other solution for that would be great!!!!

**Comment by polyrainbow** [comment]
Yes, I'm actually working right now on this feature. Stay tuned.

**Comment by infantilo** [comment]
Great to hear :-) THX

**Comment by infantilo** [comment]
Any news about this feature right now?

**Comment by polyrainbow** [comment]
The problem is, that the internal mechanisms of MediaDevices.getUserMedia() [1] seem to normalize the live input signal, which prevents a proper loudness measurement. So far, I have no idea on how to solve this. Related SO topic: [2]

[1] https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia 

[2] https://stackoverflow.com/questions/26075224/level-normalization-of-microphone-input

**Comment by infantilo** [comment]
Had the same problem on some android devices, where the mic input was only
available normalized. During my work on a RTC voip solution for my sport
commentators, the pc's audio input was not leveled/normalized at all. I'm
coding this for chrome only.
Do you have an "working" exmple for the live input right now to try?

2017-07-20 13:07 GMT+02:00 Sebastian Zimmer <notifications@github.com>:

**Comment by infantilo** [comment]
Fount that sice june 2nd two new microphone constraints got standardized, available now with adapter.js or (Firefox Nightly):

{ autoGainControl: true, noiseSuppression: false }
